File: Easy/affidamento/meta_affidamentodefaultview/meta_affidamentodefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_affidamentodefaultview
{
    public class Meta_affidamentodefaultview : Meta_easydata 
	{
        public Meta_affidamentodefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "affidamentodefaultview") {
				Name = "Affidamento";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"aa","idaffidamento","idattivform","idcanale","idcorsostudio","iddidprog","iddidproganno","iddidprogcurr","iddidprogori","iddidprogporzanno"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "registry_docenti_title", "Docente", nPos++);
						DescribeAColumn(T, "affidamentokind_title", "Tipologia", nPos++);
						DescribeAColumn(T, "affidamento_riferimento", "Docente di riferimento per il canale", nPos++);
						DescribeAColumn(T, "erogazkind_title", "Tipo di erogazione", nPos++);
						DescribeAColumn(T, "affidamento_freqobbl", "Frequenza obbligatoria", nPos++);
						DescribeAColumn(T, "affidamento_gratuito", "Gratuito", nPos++);
						DescribeAColumn(T, "affidamento_start", "Inizio", nPos++);
						DescribeAColumn(T, "affidamento_stop", "Fine", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/affidamento/meta_affidamentodefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_affidamentodefaultview() {
        MetaData.apply(this, ["affidamentodefaultview"]);
        this.name = 'meta_affidamentodefaultview';
    }

    meta_affidamentodefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_affidamentodefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'registrydocenti_title', 'Docente', null, 1300, 101);
						this.describeAColumn(table, 'affidamentokind_title', 'Tipologia', null, 2200, 50);
						this.describeAColumn(table, 'affidamento_riferimento', 'Docente di riferimento per il canale', null, 3000, null);
						this.describeAColumn(table, 'erogazkind_title', 'Tipo di erogazione', null, 4200, 50);
						this.describeAColumn(table, 'affidamento_freqobbl', 'Frequenza obbligatoria', null, 5000, null);
						this.describeAColumn(table, 'affidamento_gratuito', 'Gratuito', null, 6000, null);
						this.describeAColumn(table, 'affidamento_start', 'Inizio', null, 7000, null);
						this.describeAColumn(table, 'affidamento_stop', 'Fine', null, 8000, null);
						this.describeAColumn(table, 'XXaffidamentocaratteristica', 'Caratteristiche dell\'affidamento', null, 9000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["aa", "idcanale", "iddidprog", "idattivform", "iddidprogori", "idaffidamento", "idcorsostudio", "iddidproganno", "iddidprogcurr", "iddidprogporzanno"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "affidamento_riferimento asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('affidamentodefaultview', new meta_affidamentodefaultview('affidamentodefaultview'));

	}());
